lib/bugsnag/helpers.rb

```diff
@@ -60,6 +83,45 @@ def self.deep_merge!(l_hash, r_hash)
 
     TRUNCATION_INFO = '[TRUNCATED]'
 
+    # Trim stacktrace code out if they're too large, oldest functions first
+    def self.trim_stacktrace_code(payload, threshold)
+      return payload unless payload.is_a?(Hash) and payload[:events].respond_to?(:map)
+      payload[:events].map do |event|
+        event[:exceptions].map do |exception|
+          initial_size = get_payload_length(exception[:stacktrace])
```

Comment:
just do `exception[:stacktrace].each` and pop the code from each entry. No need for all of these checks and breaks or using `downto`.

Reply:
:+1: